CMakeLists.txt

```diff
@@ -97,16 +97,20 @@ endif ()
 # Avoid using O3
 set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG")
 
-check_cxx_compiler_flag(-march=native HAS_MARCH)
+if (NOT BUILD_ARCH)
```

Comment:
Make it an explicit option with default as "native"

Reply:
Should we explicitly fail to continue if the BUILD_ARCH fails during the compiler test on both -march= and -mcpu=?